core/workspace_comment_render_svg.js

```diff
@@ -0,0 +1,459 @@
+};
+
+/**
+ * Callback function triggered when the comment has resized.
+ * Resize the text area accordingly.
+ * @private
+ */
+Blockly.WorkspaceCommentSvg.prototype.resizeComment_ = function() {
+  var size = this.getHeightWidth();
+  var topOffset = Blockly.WorkspaceCommentSvg.TOP_OFFSET;
+  var textOffset = Blockly.WorkspaceCommentSvg.TEXTAREA_OFFSET * 2;
+
+  this.foreignObject_.setAttribute('width',
+      size.width);
+  this.foreignObject_.setAttribute('height',
+      size.height - topOffset);
+  if (this.RTL) {
+    this.foreignObject_.setAttribute('x',
+        -size.width);
+  }
+  this.textarea_.style.width =
+      (size.width - textOffset) + 'px';
+  this.textarea_.style.height =
+      (size.height - textOffset - topOffset) + 'px';
+};
+
+/**
+ * Set size
+ * @param {number} width width of the container
+ * @param {number} height height of the container
+ * @private
+ */
+Blockly.WorkspaceCommentSvg.prototype.setSize_ = function(width, height) {
+  // Minimum size of a comment.
+  width = Math.max(width, 45);
+  height = Math.max(height, 20 + Blockly.WorkspaceCommentSvg.TOP_OFFSET);
+  this.width_ = width;
+  this.height_ = height;
+  this.svgRect_.setAttribute('width', width);
+  this.svgRect_.setAttribute('height', height);
+  this.svgRectTarget_.setAttribute('width', width);
+  this.svgRectTarget_.setAttribute('height', height);
+  this.svgHandleTarget_.setAttribute('width', width);
+  this.svgHandleTarget_.setAttribute('height', Blockly.WorkspaceCommentSvg.TOP_OFFSET);
+  if (this.RTL) {
+    this.svgRect_.setAttribute('transform', 'scale(-1 1)');
+    this.svgRectTarget_.setAttribute('transform', 'scale(-1 1)');
+  }
+
+  var resizeSize = Blockly.WorkspaceCommentSvg.RESIZE_SIZE;
+  if (this.resizeGroup_) {
+    if (this.RTL) {
+      // Mirror the resize group.
+      this.resizeGroup_.setAttribute('transform', 'translate(' +
+        (-width + resizeSize) + ',' + (height - resizeSize) + ') scale(-1 1)');
+      this.deleteGroup_.setAttribute('transform', 'translate(' +
+        (-width + resizeSize) + ',' + (-resizeSize) + ') scale(-1 1)');
+    } else {
+      this.resizeGroup_.setAttribute('transform', 'translate(' +
+        (width - resizeSize) + ',' +
+        (height - resizeSize) + ')');
+      this.deleteGroup_.setAttribute('transform', 'translate(' +
+        (width - resizeSize) + ',' +
+        (-resizeSize) + ')');
+    }
+  }
+
+  // Allow the contents to resize.
+  this.resizeComment_();
+};
+
+/**
+ * Dispose of any rendered comment components.
+ * @private
+ */
+Blockly.WorkspaceCommentSvg.prototype.disposeInternal_ = function() {
+  this.textarea_ = null;
+  this.foreignObject_ = null;
+  this.svgRectTarget_ = null;
+  this.svgHandleTarget_ = null;
+};
+
+/**
+ * Set the focus on the text area.
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.prototype.setFocus = function() {
+  this.focused_ = true;
+  var textarea = this.textarea_;
+  this.svgRectTarget_.style.fill = "none";
```

Comment:
can this be done by toggling a class instead?

Reply:
Done, but it actually makes there be more code + in more places.  I don't have a strong opinion about the best way to do it.